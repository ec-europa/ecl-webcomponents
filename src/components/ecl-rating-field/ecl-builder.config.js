const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-rating-field-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-rating-field-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-rating-field-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-rating-field-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
  ],
};
