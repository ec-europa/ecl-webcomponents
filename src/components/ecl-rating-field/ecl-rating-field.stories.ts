const getArgs = () => {
  return {
    label: 'Please rate',
    required: true,
    disabled: false,
    helperText: `This is the group's helper text.`,
    requiredText: '*',
    invalidText: 'This is the error message',
    invalidIcon: 'error',
    optionalText: '(optional)',
  };
};

const getArgTypes = () => {
  return {
    required: {
      type: { name: 'boolean' },
      description: 'Required input',
    },
    disabled: {
      type: { name: 'boolean' },
      description: 'Disabled input',
    },
    label: {
      type: { name: 'string' },
      description: 'Legend of the form group',
    },
    requiredText: {
      name: 'required-text',
      type: { name: 'string' },
      description: 'Text to show when required',
    },
    optionalText: {
      name: 'optional-text',
      type: { name: 'string' },
      description: 'Text to show when optional',
    },
    helperText: {
      name: 'helper-text',
      type: { name: 'string' },
      description: 'Helper text',
    },
    invalidText: {
      name: 'invalid-text',
      type: { name: 'string' },
      description: 'Error message',
    },
    invalidIcon: {
      name: 'invalid-icon',
      type: { name: 'string' },
      description: 'Error icon',
    },
  };
};

export default {
  title: 'Components/rating field',
};

const Template = (args) => 
  `<ecl-form-group
    helper-text="${args.helperText}"
    helper-id="helper-id"
    invalid-text="${args.invalidText}"
    invalid-icon="${args.invalidIcon}"
    optional-text="${args.optionalText}"
    label="${args.label}"
    required=${args.required}
    disabled="${args.disabled}"
  >
    <ecl-rating-field
      disabled=${args.disabled}
      required=${args.required}
    >
      <ecl-rating-star
        theme="${args.theme}"
        value="1"
        item-id="item-id-1"
        label="1 star"
        disabled=${args.disabled}
        required=${args.required}
      >
      </ecl-rating-star>
      <ecl-rating-star
        theme="${args.theme}"
        value="2"
        item-id="item-id-2"
        label="2 stars"
        disabled=${args.disabled}
        required=${args.required}
      >
      </ecl-rating-star>
      <ecl-rating-star
        theme="${args.theme}"
        value="3"
        item-id="item-id-3"
        label="3 stars"
        disabled=${args.disabled}
        required=${args.required}
      >
      </ecl-rating-star>
      <ecl-rating-star
        theme="${args.theme}"
        value="4"
        item-id="item-id-4"
        label="4 stars"
        disabled=${args.disabled}
        required=${args.required}
      >
      </ecl-rating-star>
      <ecl-rating-star
        theme="${args.theme}"
        value="5"
        item-id="item-id-5"
        label="5 stars"
        disabled=${args.disabled}
        required=${args.required}
      >
      </ecl-rating-star>
    </ecl-rating-field>
  </ecl-form-group>`;

export const RatingField = Template.bind({});
RatingField.storyName = 'default';
RatingField.args = getArgs();
RatingField.argTypes = getArgTypes();