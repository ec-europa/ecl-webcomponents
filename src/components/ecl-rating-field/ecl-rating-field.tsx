import { Component, h, Prop, Element } from '@stencil/core';

@Component({
  tag: 'ecl-rating-field',
  styleUrls: {
    ec: './build/styles/ecl-rating-field-ec.css',
    eu: './build/styles/ecl-rating-field-eu.css',
  },
  shadow: false,
  scoped: true,
})
export class EclRatingField {
  @Element() el: HTMLElement;
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() disabled: boolean = false;
  @Prop() required: boolean = false;

  componentDidRender() {
    const stars = this.el.querySelectorAll('.ecl-rating-field__star');
    this.el.querySelector('.ecl-rating-field').innerHTML = '';
    this.el.querySelector('.ecl-rating-field').append(...stars);
  }

  getClass(): string {
    const styleClasses = [
      `ecl-rating-field`,
      this.styleClass
    ];

    if (this.disabled) {
      styleClasses.push('ecl-rating-field--disabled');
    }

    return styleClasses.join(' ');
  }

  render() {
    return (
      <div class={this.getClass()}>
        <slot></slot>
      </div>
    );
  }
}