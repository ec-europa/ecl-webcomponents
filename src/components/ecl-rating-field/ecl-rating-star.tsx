import { Component, h, Prop } from '@stencil/core';
import Fragment from 'stencil-fragment';

@Component({
  tag: 'ecl-rating-star',
  styleUrls: {
    ec: './build/styles/ecl-rating-field-ec.css',
    eu: './build/styles/ecl-rating-field-eu.css',
  },
  shadow: false,
})
export class EclRatingStar {
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() itemId: string;
  @Prop() value: string;
  @Prop() name: string;
  @Prop() checked: boolean = false;
  @Prop() required: boolean = false;
  @Prop() disabled: boolean = false;
  @Prop() icon: string = 'star-outline';
  @Prop() iconFilled: string = 'star-filled';
  @Prop() label: string;

  render() {
    return (
      <Fragment>
        <input
          id={this.itemId}
          class="ecl-rating-field__input ecl-rating-field__star"
          type="radio"
          name={this.name}
          value={this.value}
          checked={this.checked}
          required={this.required}
          disabled={this.disabled}
        />
        <label 
          class="ecl-rating-field__label ecl-rating-field__star" 
          htmlFor={this.itemId}
        >
        { this.label ?
          <span class="ecl-u-sr-only">{ this.label }</span> : '' }
          <ecl-icon
            icon={this.iconFilled}
            styleClass="ecl-rating-field__icon-filled"
            size="m"
          >
          </ecl-icon>
          <ecl-icon
            icon={this.icon}
            styleClass="ecl-rating-field__icon-outline"
            size="m"
          >
          </ecl-icon>
        </label>
      </Fragment>
    );
  }
}