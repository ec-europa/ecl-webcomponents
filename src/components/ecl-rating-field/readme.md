# ecl-rating-field-item



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description | Type      | Default          |
| ------------ | ------------- | ----------- | --------- | ---------------- |
| `checked`    | `checked`     |             | `boolean` | `false`          |
| `disabled`   | `disabled`    |             | `boolean` | `false`          |
| `icon`       | `icon`        |             | `string`  | `'star-outline'` |
| `iconFilled` | `icon-filled` |             | `string`  | `'star-filled'`  |
| `itemId`     | `item-id`     |             | `string`  | `undefined`      |
| `label`      | `label`       |             | `string`  | `undefined`      |
| `name`       | `name`        |             | `string`  | `undefined`      |
| `required`   | `required`    |             | `boolean` | `false`          |
| `styleClass` | `style-class` |             | `string`  | `undefined`      |
| `theme`      | `theme`       |             | `string`  | `'ec'`           |
| `value`      | `value`       |             | `string`  | `undefined`      |


## Dependencies

### Depends on

- [ecl-icon](../ecl-icon)

### Graph
```mermaid
graph TD;
  ecl-rating-star --> ecl-icon
  style ecl-rating-star fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
