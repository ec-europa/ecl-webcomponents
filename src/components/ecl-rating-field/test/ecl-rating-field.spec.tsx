import { newSpecPage } from '@stencil/core/testing';
import { EclRatingField } from '../ecl-rating-field';
import { EclRatingStar } from '../ecl-rating-star';

describe('ecl-rating-field', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclRatingField, EclRatingStar],
      html: `<ecl-rating-field
        required
        theme="ec"
      >
        <ecl-rating-star
          theme="ec"
          value="1"
          item-id="item-id-1"
          label="1 stars"
          required
        >
        </ecl-rating-star>
        <ecl-rating-star
          theme="ec"
          value="2"
          item-id="item-id-2"
          label="2 stars"
          required
        >
        </ecl-rating-star>
        <ecl-rating-star
          theme="ec"
          value="3"
          item-id="item-id-3"
          label="3 stars"
          required
        >
        </ecl-rating-star>
        <ecl-rating-star
          theme="ec"
          value="4"
          item-id="item-id-4"
          label="4 stars"
          required
        >
        </ecl-rating-star>
        <ecl-rating-star
          theme="ec"
          value="5"
          item-id="item-id-5"
          label="5 stars"
          required
        >
        </ecl-rating-star>
      </ecl-rating-field>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});