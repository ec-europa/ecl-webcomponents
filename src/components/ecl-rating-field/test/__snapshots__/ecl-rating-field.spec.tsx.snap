// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-rating-field renders 1`] = `
<ecl-rating-field required="" theme="ec">
  <!---->
  <div class="ecl-rating-field">
    <input class="ecl-rating-field__input ecl-rating-field__star" id="item-id-1" required="" type="radio" value="1">
    <label class="ecl-rating-field__label ecl-rating-field__star" htmlfor="item-id-1">
      <span class="ecl-u-sr-only">
        1 stars
      </span>
      <ecl-icon icon="star-filled" size="m" styleclass="ecl-rating-field__icon-filled"></ecl-icon>
      <ecl-icon icon="star-outline" size="m" styleclass="ecl-rating-field__icon-outline"></ecl-icon>
    </label>
    <input class="ecl-rating-field__input ecl-rating-field__star" id="item-id-2" required="" type="radio" value="2">
    <label class="ecl-rating-field__label ecl-rating-field__star" htmlfor="item-id-2">
      <span class="ecl-u-sr-only">
        2 stars
      </span>
      <ecl-icon icon="star-filled" size="m" styleclass="ecl-rating-field__icon-filled"></ecl-icon>
      <ecl-icon icon="star-outline" size="m" styleclass="ecl-rating-field__icon-outline"></ecl-icon>
    </label>
    <input class="ecl-rating-field__input ecl-rating-field__star" id="item-id-3" required="" type="radio" value="3">
    <label class="ecl-rating-field__label ecl-rating-field__star" htmlfor="item-id-3">
      <span class="ecl-u-sr-only">
        3 stars
      </span>
      <ecl-icon icon="star-filled" size="m" styleclass="ecl-rating-field__icon-filled"></ecl-icon>
      <ecl-icon icon="star-outline" size="m" styleclass="ecl-rating-field__icon-outline"></ecl-icon>
    </label>
    <input class="ecl-rating-field__input ecl-rating-field__star" id="item-id-4" required="" type="radio" value="4">
    <label class="ecl-rating-field__label ecl-rating-field__star" htmlfor="item-id-4">
      <span class="ecl-u-sr-only">
        4 stars
      </span>
      <ecl-icon icon="star-filled" size="m" styleclass="ecl-rating-field__icon-filled"></ecl-icon>
      <ecl-icon icon="star-outline" size="m" styleclass="ecl-rating-field__icon-outline"></ecl-icon>
    </label>
    <input class="ecl-rating-field__input ecl-rating-field__star" id="item-id-5" required="" type="radio" value="5">
    <label class="ecl-rating-field__label ecl-rating-field__star" htmlfor="item-id-5">
      <span class="ecl-u-sr-only">
        5 stars
      </span>
      <ecl-icon icon="star-filled" size="m" styleclass="ecl-rating-field__icon-filled"></ecl-icon>
      <ecl-icon icon="star-outline" size="m" styleclass="ecl-rating-field__icon-outline"></ecl-icon>
    </label>
  </div>
</ecl-rating-field>
`;
